blockchain/process.go

```diff
@@ -202,7 +202,7 @@ func (b *BlockChain) ProcessBlock(block *btcutil.Block, flags BehaviorFlags) (bo
 	log.Tracef("Processing block %v", blockHash)
 
 	// The block must not already exist in the main chain or side chains.
-	exists, err := b.blockExists(blockHash)
+	exists, err := b.HaveBlock(blockHash)
```

Comment:
Note the comment on `HaveBlock`. We can change the behavior of `blockExists` instead of changing the behavior of `HaveBlock`

Reply:
I have changed the behaviour of blockExists and it checks and returns, whether there is block data or not